src/request_list.js

```diff
@@ -186,90 +184,140 @@ class RequestList {
      *
      * @returns {Promise}
      */
-    initialize() {
+    async initialize() {
         if (this.isLoading) {
             throw new Error('RequestList sources are already loading or were loaded.');
         }
         this.isLoading = true;
 
+        let state;
+        let sources;
+        if (this.initialState) {
```

Comment:
I am not sure if this workflow is correct. I would handle state and sources separately based on `this.persistStateKey`, `this.persistSourcesKey` variables.

If one is defined then try to get value from kv-store. If it's not there use initial one and persist it in kv-store.